MainWindow.cpp

```diff
@@ -564,7 +564,13 @@ void MainWindow::CreateResource(TypeCase typeCase) {
   refl->MutableMessage(&child, field);
 
   // find a unique name for the new resource
-  child.set_name(resourceMap->CreateResourceName(&child).toStdString());
+  QString resourceName = resourceMap->CreateResourceName(&child);
+  child.set_name(resourceName.toStdString());
+
+  // add new resource with created name, helps in creating another unique name
+  auto messageModelRef = MessageModel(ProtoModel::NonProtoParent{this}, &child);
+  resourceMap->AddResource(typeCase, resourceName, &messageModelRef);

```

Comment:
`messageModelRef` is a temporary in this context; this reference will die after this call to `CreateResource` terminates. Doesn't this cause instability in the IDE?